Jint/Runtime/TypeConverter.cs

```diff
@@ -634,24 +638,47 @@ public static void CheckObjectCoercible(Engine engine, JsValue o)
                         }
                         else
                         {
-                            // no can do
-                            perfectMatch = false;
-                            break;
+                            if (paramType.IsAssignableFrom(argType))
+                            {
+                                score -= 1;
+                            }
+                            else
+                            {
+                                if (argType.GetMethods(BindingFlags.Public | BindingFlags.Static)
```

Comment:
Is there a test for that and the `scrore` thing to understand what is happening?

Reply:
Basically this method finds the best match when there are multiple methods with the same name and number of arguments. Before my changes, this code was checking for a method that was perfectly matching to the passed arguments. If it could not find a perfect match, it would choose the first method it encountered (and this would cause an error in most cases).

My code gives each method a penalty based on how close each argument is to the parameter it is going to be casted to. Score is divided into different levels and assigned with a number in base 10. This is kind of a heuristic algorithm.

This is demonstrated in `MethodAmbiguityTests.cs`.

To be honest, this became deeper than I initially thought. It deserves its own PR. I did not want to go overkill here. C# compiler can resolve the best method even better. But that is probably only possible with static analysis. Also when there are multiple matching methods, C# calls it an ambiguity and does not let you compile.

Also this can become a breaking change for people who use interop a lot and have methods with same names. Perfect match will still be same but when there are multiple non-matching methods, people may experience different results with this change. But in my opinion, before this change, the result would be undefined behavior.